src/Agent.Worker/JobRunner.cs

```diff
@@ -189,6 +213,29 @@ public sealed class JobRunner : AgentService, IJobRunner
                     }
                 }
 
+#if OS_WINDOWS
+                // Add script post steps.
+                // This is for internal testing and is not publicly supported. This will be removed from the agent at a later time.
+                var finallyScript = Environment.GetEnvironmentVariable("VSTS_AGENT_CLEANUP");
```

Comment:
Can we make the variable VSTS_AGENT_CLEANUP_INTERNAL_TEMP_HACK, or something more obvious that it's not externally supported?

When customers complain because we break when we remove this, I don't want our rebuttal to be "we had a comment in the source code that you shouldnt use it". I want it to be obvious that they took a dependency on something internally supported only.